src/Compilers/Core/Portable/Diagnostic/MetadataLocation.cs

```diff
@@ -40,7 +40,7 @@ public override bool Equals(object obj)
 
         public bool Equals(MetadataLocation other)
         {
-            return other != null && other._module == _module;
+            return other is { _module: _module };
```

Comment:
>_module [](start = 39, length = 7)

Not a valid pattern (not a constant)

Reply:
➡️ Treating this one as a duplicate of https://github.com/dotnet/roslyn/pull/39738#discussion_r344416577